lib/ansible/galaxy/collection/__init__.py

```diff
@@ -684,27 +412,122 @@ def install_collections(collections, output_path, apis, validate_certs, ignore_e
     :param force: Re-install a collection if it has already been installed.
     :param force_deps: Re-install a collection as well as its dependencies if they have already been installed.
     """
-    existing_collections = find_existing_collections(output_path, fallback_metadata=True)
+    existing_collections = {
+        Requirement(coll.fqcn, coll.ver, coll.src, coll.type)
+        for coll in find_existing_collections(output_path, artifacts_manager)
+    }
 
-    with _tempdir() as b_temp_path:
-        with _display_progress("Process install dependency map"):
-            dependency_map = _build_dependency_map(collections, existing_collections, b_temp_path, apis,
-                                                   validate_certs, force, force_deps, no_deps,
-                                                   allow_pre_release=allow_pre_release)
+    requested_requirements = set(
+        chain.from_iterable(
+            (
+                Requirement.from_dir_path(sub_coll, artifacts_manager)
+                for sub_coll in (
+                    artifacts_manager.
+                    get_direct_collection_dependencies(install_req).
+                    keys()
+                )
+            )
+            if install_req.is_subdirs else (install_req, )
+            for install_req in collections
+        ),
+    )
+    requested_requirements_names = {req.fqcn for req in requested_requirements}
+
+    # NOTE: Don't attempt to reevaluate already installed deps
+    # NOTE: unless `--force` is passed
+    # FIXME: debug why `--force` doesn't always work well
+    requested_requirements -= set() if force else {
```

Comment:
`--force-with-deps` implies `--force` on devel. I'm not sure if we want to fix that by checking for either, e.g. 
```suggestion
    requested_requirements -= set() if (force or force_deps) else {
```
or if we should set `force` to True in cli/galaxy.py if `--force-with-deps` was provided (might be more minimal that way, since it looks like there are a few places we check for `force`).

Reply:
I'm about to rewrite a few bits around this logic because this var is no longer passed to the resolver and we still need some fixes related to SCM that would probably need to touch this now.